dialoguekit/core/annotated_utterance.py

```diff
@@ -55,12 +55,13 @@ def get_intents(self) -> List[Intent]:
         """Returns utterance's intents."""
         return [da.intent for da in self.dialogue_acts]
 
-    def get_dialogue_act_annotations(self) -> Set[Annotation]:
+    def get_dialogue_act_annotations(self) -> List[Annotation]:
```

Comment:
I misunderstood the purpose of this method, initially I thought it's for returning the dialogue acts. But currently this does not seem to be used anywhere. Do you have a specific idea of where it'd be needed? If not, we should not have it at this point.

Reply:
This method is used in `_filter_templates` and `get_intent_annotation_specifications` in `TemplateNLG`. From my understand it is used to filter the template when the option `force_annotation` is set to true.
After a careful look, I wonder about the utility of the method `get_intent_annotation_specifications` though.

Reply:
I see. In both cases though it's only used to check if there are any dialogue acts with slot-value pairs. So perhaphs turning it into a `are_dialogue_acts_with_slot_values() -> bool` method would be sufficient?

Reply:
Sounds good